File: frontend/src/types/order.ts
import type { PageQuery } from './api'

export type OrderStatus =
  | 'PENDING'
  | 'ACCEPTED'
  | 'COOKING'
  | 'DELIVERING'
  | 'COMPLETED'
  | 'CANCELLED'

export interface OrderItem {
  id: number
  dishId: number
  dishName: string
  dishPrice: number
  quantity: number
  subtotal: number
}

export interface OrderTimelineItem {
  status: OrderStatus
  label: string
  active: boolean
  time: string | null
}

export interface OrderSummary {
  id: number
  orderNo: string
  username?: string
  totalAmount: number
  receiverName?: string
  receiverPhone?: string
  receiverAddress?: string
  status: OrderStatus
  statusText?: string
  createdTime: string
  itemCount?: number
}

export interface OrderDetail extends OrderSummary {
  userId?: number
  remark?: string
  updatedTime?: string
  items: OrderItem[]
  timeline?: OrderTimelineItem[]
}

export interface CreateOrderRequest {
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  remark?: string
}

export interface CreateOrderResponse {
  orderId: number
  orderNo: string
  totalAmount: number
  status: OrderStatus
}

export interface MyOrderQuery extends PageQuery {
  status?: OrderStatus
}

export interface AdminOrderQuery extends PageQuery {
  keyword?: string
  status?: OrderStatus
  startDate?: string
  endDate?: string
}

File: backend/src/main/java/com/njit/takeoutsystem/entity/Order.java
package com.njit.takeoutsystem.entity;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class Order {
    private Long id;
    private String orderNo;
    private Long userId;
    private String username;
    private BigDecimal totalAmount;
    private String receiverName;
    private String receiverPhone;
    private String receiverAddress;
    private String remark;
    private String status;
    private String cancelReason;
    private LocalDateTime createdTime;
    private LocalDateTime updatedTime;
    private Integer itemCount;
}
